services/collaboration/pkg/middleware/tracing.go

```diff
@@ -0,0 +1,39 @@
+package middleware
+
+import (
+	"net/http"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+func CollaborationTracingMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		wopiContext, err := WopiContextFromCtx(r.Context())
+		if err != nil {
+			// if we can't get the context, skip this middleware
+			next.ServeHTTP(w, r)
+		}
+
+		span := trace.SpanFromContext(r.Context())
+
+		wopiMethod := r.Header.Get("X-WOPI-Override")
```

Comment:
I'd love to see `X-WOPI-SessionId`, too. See also https://learn.microsoft.com/en-us/microsoft-365/cloud-storage-partner-program/rest/common-headers

Reply:
This is included now. As far as I know this is only for office365, so I'd expect it to be empty for Collabora and OnlyOffice

Reply:
OnlyOffice has it, too from what I have seen. Though I need to clarify if it's a session id that makes sense (looked like the user id to me).